src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -738,6 +777,41 @@ impl Drop for CanceledWorkunitGuard {
   }
 }
 
+#[derive(Clone)]
+struct MetricsData {
+  counters: Arc<Mutex<HashMap<SpanId, HashMap<Metric, u64>>>>,
+}
+
+impl Default for MetricsData {
+  fn default() -> MetricsData {
+    MetricsData {
+      counters: Arc::new(Mutex::new(HashMap::new())),
+    }
+  }
+}
+
+impl MetricsLike for WorkunitStore {
```

Comment:
Nit: the trait feels like overkill... I tend to wait to extract interfaces until they're necessary for testing.

Reply:
Whereas I tend to write the interface first to keep the set of operations concise in my mind especially if I do not know where the operations will end up. Or I wrote the trait during all the iteration on this PR once told that there needed to be a "metrics setter" type. I forget.

I will refactor this into work unit store.